src/hyp3_opera_rtc/dem.py

```diff
@@ -54,38 +87,91 @@ def check_antimeridean(poly: Polygon) -> list[Polygon]:
+    # update cropping coordinates to not exceed the input DEM bounding box
+    input_x_min, xres, _, input_y_max, _, yres = ds.GetGeoTransform()
+    length = ds.GetRasterBand(1).YSize
+    width = ds.GetRasterBand(1).XSize
+
+    # Snap edge coordinates using the DEM pixel spacing
+    # (xres and yres) and starting coordinates (input_x_min and
+    # input_x_max). Maximum values are rounded using np.ceil
+    # and minimum values are rounded using np.floor
+    x_min, y_min, x_max, y_max = bounds
+    snapped_x_min = snap_coord(x_min, xres, input_x_min, np.floor)
+    snapped_x_max = snap_coord(x_max, xres, input_x_min, np.ceil)
+    snapped_y_min = snap_coord(y_min, yres, input_y_max, np.floor)
+    snapped_y_max = snap_coord(y_max, yres, input_y_max, np.ceil)
+
+    input_y_min = input_y_max + length * yres
+    input_x_max = input_x_min + width * xres
+
+    adjusted_x_min = max(snapped_x_min, input_x_min)
+    adjusted_x_max = min(snapped_x_max, input_x_max)
+    adjusted_y_min = max(snapped_y_min, input_y_min)
+    adjusted_y_max = min(snapped_y_max, input_y_max)
+
+    try:
+        gdal.Translate(
+            output_path, ds, format='GTiff', projWin=[adjusted_x_min, adjusted_y_max, adjusted_x_max, adjusted_y_min]
+        )
+    except RuntimeError as err:
+        if 'negative width and/or height' in str(err):
+            gdal.Translate(output_path, ds, format='GTiff', projWin=[x_min, y_max, x_max, y_min])
+        else:
+            raise
+
+    # stage_dem.py takes a bbox as an input. The longitude coordinates
+    # of this bbox are unwrapped i.e., range in [0, 360] deg. If the
+    # bbox crosses the anti-meridian, the script divides it in two
+    # bboxes neighboring the anti-meridian. Here, x_min and x_max
+    # represent the min and max longitude coordinates of one of these
+    # bboxes. We Add 360 deg if the min longitude of the downloaded DEM
+    # tile is < 180 deg i.e., there is a dateline crossing.
+    # This ensures that the mosaicked DEM VRT will span a min
+    # range of longitudes rather than the full [-180, 180] deg
+    sr = osr.SpatialReference(ds.GetProjection())
+    epsg_str = sr.GetAttrValue('AUTHORITY', 1)
+
+    if x_min <= -180.0 and epsg_str == '4326':
+        ds = gdal.Open(output_path, gdal.GA_Update)
+        geotransform = list(ds.GetGeoTransform())
+        geotransform[0] += 360.0
+        ds.SetGeoTransform(tuple(geotransform))
+
+
+def download_opera_dem_for_footprint(outfile: Path, bounds: tuple[float, float, float, float]) -> None:
+    """Download a DEM from the specified S3 bucket.
+
+    Params:
+        polys: List of shapely polygons.
+        outfile: Path to the where the output DEM file is to be staged.
+    """
+    poly = polygon_from_bounds(bounds)
+    polys = check_antimeridean(poly)
+    dem_list = []
+
+    os.environ['GDAL_HTTP_COOKIEJAR'] = 'cookies.txt'
+    os.environ['GDAL_HTTP_COOKIEFILE'] = 'cookies.txt'
+    os.environ['GDAL_DISABLE_READDIR_ON_OPEN'] = 'EMPTY_DIR'
+
```

Comment:
I'd prefer to see `hyp3lib.util.GDALConfigManager` and `tempfile.NamedTemporaryFile` used here, e.g.:

```
    with tempfile.NamedTemporaryFile() as cookie_file:
        with hyp3lib.util.GDALConfigManager(
            GDAL_HTTP_COOKIEJAR=cookie_file.name,
            GDAL_HTTP_COOKIEFILE=cookie_file.name,
            GDAL_DISABLE_READDIR_ON_OPEN='EMPTY_DIR',
        ):
```